### DISCUSSION TITLE
Policy:Subnets should have a Network Security Group

### DISCUSSION THREAD

**Discussion by bagelsandcoffee** [Azure/Enterprise-Scale#1031] [Ideas] [answered]
Hello ,

This policy Subnets should have a Network Security Group is deployed when VWAN is selected along with other parameter on 
Deploy Enterprise-Scale with Azure VWAN. I was testing the policies and looks like  the policy denies creation of a VNET completely 
 I am wondering if the creation  is completely denied how would a NSG be attached and how new vents will be created I understand there is audit option but the deployment creates assignment with "deny".

Thankx

**Comment by jtracey93** [comment]
Hi @bagelsandcoffee,

This policy is assigned to the `identity` and `landing zones` management groups, as detailed [here](https://github.com/Azure/Enterprise-Scale/blob/main/docs/ESLZ-Policies.md).

The policy only looks at each subnet and checks there is an NSG assigned. If not, the `deny` action will take effect. It does not block the creation of a VNET.

Hope that helps, let me know if you need anything else or have more questions

**Reply by bagelsandcoffee** [reply]
Thankx @jtracey93   for the reply .I will test it out again and let you know

**Comment by bagelsandcoffee** [comment, marked as answer]
Thankx for resources was very helpful.I redeployed the solution .Same Problem.Since while creating a VNet there is no option create a NSG i cant create one  and hence the error.
@jtracey93 
![image](https://user-images.githubusercontent.com/89396868/187426527-af5e70b9-8977-4fd6-9d7d-79c8eb9fdc70.png)
![image](https://user-images.githubusercontent.com/89396868/187426558-ca85903d-6c07-45d4-9525-7ecb1ed14b3b.png)

**Reply by jtracey93** [reply]
Ah okay, you are using the portal. This now makes sense, and in-fact is already logged as a bug here #972 

The VNET portal team are aware and are working on it. However for now you can create a blank VNET via a [Bicep file](https://docs.microsoft.com/en-us/azure/virtual-network/quick-create-bicep?tabs=CLI) for example or [PowerShell](https://docs.microsoft.com/en-us/azure/virtual-network/quick-create-powershell) and then you can use the portal expeirence from there.

You could also create a [template spec](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/template-specs?tabs=azure-powershell) for a blank VNET with no subnets, to get around this also today. Then you can use the portal as normal to create new NSG and subnets once the blank VNET is created.

VNET Portal Engineering are working on this though, for an updated portal experience for VNETs that is due soon (no ETA to share im afraid, but we have seen an early preview 👍)

**Reply by bagelsandcoffee** [reply]
Sweet .Thankx for reply makes sense
